packages/nx/src/command-line/init/init-v2.ts

```diff
@@ -0,0 +1,184 @@
+import { prerelease } from 'semver';
+import { output } from '../../utils/output';
+import { getPackageManagerCommand } from '../../utils/package-manager';
+import { generateDotNxSetup } from './implementation/dot-nx/add-nx-scripts';
+import { runNxSync } from '../../utils/child-process';
+import { readJsonFile } from '../../utils/fileutils';
+import { nxVersion } from '../../utils/versions';
+import {
+  addDepsToPackageJson,
+  askAboutNxCloud,
+  createNxJsonFile,
+  runInstall,
+  updateGitIgnore,
+} from './implementation/utils';
+import { prompt } from 'enquirer';
+import { execSync } from 'child_process';
+
+export interface InitArgs {
+  addE2e: boolean;
+  force: boolean;
+  integrated: boolean;
+  interactive: boolean;
+  vite: boolean;
+  nxCloud?: boolean;
+  cacheable?: string[];
+  useDotNxInstallation?: boolean;
+}
+
+export async function initHandler(options: InitArgs) {
+  const version =
+    process.env.NX_VERSION ?? (prerelease(nxVersion) ? 'next' : 'latest');
+  if (process.env.NX_VERSION) {
+    output.log({ title: `Using version ${process.env.NX_VERSION}` });
+  }
+  if (existsSync('package.json') && !options.useDotNxInstallation) {
+    const repoRoot = process.cwd();
+    const cacheableOperations: string[] = [];
+    createNxJsonFile(repoRoot, [], cacheableOperations, {});
+
+    const pmc = getPackageManagerCommand();
+
+    updateGitIgnore(repoRoot);
+
+    const plugins = await detectPlugins(options);
+
+    const useNxCloud =
+      options.nxCloud ??
+      (options.interactive ? await askAboutNxCloud() : false);
+
+    addDepsToPackageJson(repoRoot, plugins);
+
+    output.log({ title: '📦 Installing Nx' });
+
+    runInstall(repoRoot, pmc);
+
+    if (plugins) {
+      output.log({ title: '🔨 Configuring plugins' });
+      for (const plugin of plugins) {
+        execSync(
+          `${pmc.exec} nx g ${plugin}:init --skipPackageJson --no-interactive`,
+          {
+            stdio: [0, 1, 2],
+            cwd: repoRoot,
+          }
+        );
+      }
+    }
+
+    if (useNxCloud) {
+      output.log({ title: '🛠️ Setting up Nx Cloud' });
+      execSync(
+        `${pmc.exec} nx g nx:connect-to-nx-cloud --installationSource=nx-init-pcv3 --quiet --no-interactive`,
+        {
+          stdio: [0, 1, 2],
+          cwd: repoRoot,
+        }
+      );
+    }
+  } else {
+    if (process.platform !== 'win32') {
+      console.log(
+        'Setting Nx up installation in `.nx`. You can run nx commands like: `./nx --help`'
+      );
+    } else {
+      console.log(
+        'Setting Nx up installation in `.nx`. You can run nx commands like: `./nx.bat --help`'
+      );
+    }
+    generateDotNxSetup(version);
+    // invokes the wrapper, thus invoking the initial installation process
+    runNxSync('');
+  }
+}
+
+async function detectPlugins(options: InitArgs): Promise<undefined | string[]> {
+  const files = ['package.json'].concat(
+    sync('{apps,packages,libs}/**/*/package.json')
```

Comment:
why not just a general glob under root dir? excluding `node_modules`? i mean to say, someone may have any type of setup (any dir name), no?

Reply:
Solved by using the Rust glob util instead, which will respect `.gitignore` files.